### PR TITLE
2973-feat: Skeleton Loading Added

### LINKED ISSUE #2973: Set a skeleton loading for the show page
# Current behavior

The page turns completely gray while loading the content:

https://github.com/twentyhq/twenty/assets/19412894/cd2bf961-a58b-4254-902c-7d907b4bdcf0

# Desired behavior

Display the containers so the page doesn't feel empty while loading:

image

### PR DESCRIPTION
**Pull request by Kanav-Arora**
Closes #2973 

I found the issue but was unable to reproduce it, but I have made some changes if they work.

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/components/RecordShowPage.tsx [hunk 1]
```diff
@@ -55,11 +56,14 @@ export const RecordShowPage = () => {
     entityFieldsFamilyState(objectRecordId ?? ''),
   );
 
+  const [fetching, setFetching] = useState(true);
+
   const { record } = useFindOneRecord({
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/object-record/components/RecordShowPage.tsx:61]: Could you use the loading property from the Apollo query hook instead ?

**Review by lucasbordeau — DISMISSED** [review]

**Comment by Kanav-Arora** [comment]
Hi @lucasbordeau 
Done with the changes please check

File: packages/twenty-front/src/modules/object-record/components/RecordShowPage.tsx [hunk 2]
```diff
@@ -197,73 +197,84 @@ export const RecordShowPage = () => {
...
-                    : objectMetadataItem?.nameSingular === 'person'
-                      ? 'Person'
-                      : 'Custom',
-              }}
-              timeline
-              tasks
-              notes
-              emails
-            />
+            {loading === false ? (
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/components/RecordShowPage.tsx:200]: !loading

File: packages/twenty-front/src/modules/ui/layout/show-page/components/ShowPageLeftContainer.tsx [hunk 3]
```diff
@@ -38,7 +38,7 @@ const StyledIntermediateContainer = styled.div`
 `;
 
 export type ShowPageLeftContainerProps = {
-  children: ReactElement[];
+  children?: ReactElement[];
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/layout/show-page/components/ShowPageLeftContainer.tsx:41]: this feels wrong, I think we should force containers to have children

File: packages/twenty-front/src/modules/object-record/components/RecordShowPage.tsx [hunk 4]
```diff
@@ -197,73 +197,84 @@ export const RecordShowPage = () => {
...
+                  }}
+                  timeline
+                  tasks
+                  notes
+                  emails
+                />
+              </>
+            ) : (
+              <>
+                <ShowPageLeftContainer />
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/components/RecordShowPage.tsx:274]: `<ShowPageLeftContainer></ShowPageLeftContainer>`

File: packages/twenty-front/src/modules/ui/layout/show-page/components/ShowPageRightContainer.tsx [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react-hooks/rules-of-hooks */
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/layout/show-page/components/ShowPageRightContainer.tsx:1]: why do we need this? Feels wrong :)

**Review by charlesBochet — COMMENTED** [review]
@Kanav-Arora, I have left additional comments, then we can merge!

**Comment by charlesBochet** [comment]
@Kanav-Arora any update on this one? :)

**Comment by Kanav-Arora** [comment]
Hi @charlesBochet on it
Sorry I was busy somewhere, pushing changes in an hour

**Comment by Kanav-Arora** [comment]
Hi @charlesBochet 
Done

**Review by charlesBochet — APPROVED** [review]
@Kanav-Arora I have removed` if (!record) return <></>;` on line 132 which was making your fix not actually applied. And I have fixed the consequences of this.
Thanks for the help, as always!

**Pull request merged**